tests/net/lib/mqtt_sn_packet/src/mqtt_sn_packet.c

```diff
@@ -536,7 +536,7 @@ static ZTEST(mqtt_sn_packet, test_mqtt_packet_encode)
 				"Expected data");
 		LOG_HEXDUMP_DBG(msg.data, msg.len, "Encoded data");
 		zassert_equal(msg.len, encode_tests[i].expectedsz,
-			      "Unexpected data size %zu (%zu)", msg.len,
+			      "Unexpected data size %zu (%zu)", (size_t)msg.len,
```

Comment:
If I'm seeing this correctly, `msg` is a `net_buf_simple` and its member `len` is neither a `size_t` nor an `int` but rather an `uint16_t`. I would propose to fix the formatting to the correct type instead of casting.